web/api/v1/api.go

```diff
@@ -81,6 +82,33 @@ const (
 	errorNotFound    errorType = "not_found"
 )
 
+// partialResponseStrategy controls the behaviour of given server API on how to treat temporary unavailability of remote components.
+// For example configured remote read APIs in case of Prometheus.
+// TODO(bwplotka): Move to storage and pass through hints so we can abort early if chosen.
+type partialResponseStrategy string
+
+const (
+	// partialResponseStrategyWarn strategy tells server that inconsistent results given unavailability of remote components are
+	// acceptable as long as warnings are provided in response.
+	partialResponseStrategyWarn = "warn"
+	// partialResponseStrategyAbort strategy tells server that inconsistent results given unavailability of remote components are
+	// not acceptable and such request should fail.
+	partialResponseStrategyAbort = "abort"
+)
+
+func (s *partialResponseStrategy) Parse(value string) error {
+	*s = partialResponseStrategyWarn // Default.
+	if value != "" {
+		switch v := strings.ToLower(value); v {
```

Comment:
We use case sensitive URL parameters everywhere else, so this should be the same.